sphinx/directives/other.py

```diff
@@ -106,9 +107,8 @@ def parse_content(self, toctree: addnodes.toctree) -> List[Node]:
                     toctree['entries'].append((None, docname))
                     toctree['includefiles'].append(docname)
                 if not docnames:
-                    ret.append(self.state.document.reporter.warning(
-                        'toctree glob pattern %r didn\'t match any documents'
-                        % entry, line=self.lineno))
+                    logger.warning(__('toctree glob pattern %r didn\'t match any documents'),
```

Comment:
Out of curiosity, what are the implications of using `logger.warning` rather than `document.reporter.warning`, if any?

Reply:
There are some advantages of `sphinx.util.logging`:

* It can be used even if no reporter object.
* It supports additional features like log-type, location, and so on.

In this case, I used it to make this log message suppressible.